ee/tools/puppet/fleetdm/spec/functions/preassign_profile_spec.rb

```diff
@@ -26,31 +27,31 @@
   it 'performs an API call to Fleet with the right parameters' do
     expect(fleet_client_mock)
       .to receive(:get_host_by_identifier)
-      .with(device_uuid, 'production')
+      .with(device_uuid, rspec_puppet_env)
       .and_return({ 'error' => '', 'body' => host_response })
     expect(fleet_client_mock)
       .to receive(:get_host_profiles)
-      .with(host_response['host']['id'], 'production')
+      .with(host_response['host']['id'], rspec_puppet_env)
       .and_return({ 'error' => '', 'body' => { 'profiles' => [] } })
     expect(fleet_client_mock)
       .to receive(:preassign_profile)
-      .with(run_identifier, device_uuid, template, group, ensure_profile, 'production')
+      .with(run_identifier, device_uuid, template, group, ensure_profile, rspec_puppet_env)
       .and_return({ 'error' => '' })
     is_expected.to run.with_params(profile_identifier, device_uuid, template, group, ensure_profile)
   end
 
   it 'has default values for `group` and `ensure`' do
     expect(fleet_client_mock)
       .to receive(:get_host_by_identifier)
-      .with(device_uuid, 'production')
+      .with(device_uuid, rspec_puppet_env)
       .and_return({ 'error' => '', 'body' => host_response })
     expect(fleet_client_mock)
       .to receive(:get_host_profiles)
-      .with(host_response['host']['id'], 'production')
+      .with(host_response['host']['id'], rspec_puppet_env)
       .and_return({ 'error' => '', 'body' => { 'profiles' => [] } })
     expect(fleet_client_mock)
       .to receive(:preassign_profile)
-      .with(run_identifier, device_uuid, template, 'default', 'present', 'production')
+      .with(run_identifier, device_uuid, template, 'default', 'present', rspec_puppet_env)
       .and_return({ 'error' => '' })
     is_expected.to run.with_params(profile_identifier, device_uuid, template)
   end
```

Comment:
Do we want to add more coverage in this PR for the TODO below?

Reply:
ah, thanks! I forgot about those. I'm thinking on sending a separate PR so we get the chance to test/tweak the CI workflow :D